### PR TITLE
Ref python

### PR DESCRIPTION
**Pull request by RobertGemmaJr**
Fixes the communication between Python and Unity and cleans up the logging

- xsd now asserts that `duration` attributes are >= 0
- Wraps `CLI.Main` in a try/catch block to prevent Unity from crashing
- Adds Logging throughout the Unity translation
- Unity scene is deleted if an exception occurred during translation
- Renames `ProjectPath` as `XmlPath`

### CODE REVIEW

File: unity/CAVE/Assets/Editor/CLI.cs [hunk 1]
```diff
@@ -283,9 +296,16 @@ private static void SetLinkActions()
...
+                // Prepending "LOG:" will print the line to the screen (Python script)
+                string color = type switch
+                {
+                    LogType.Log => "white",
+                    LogType.Warning => "yellow",
+                    LogType.Error => "red",
+                    LogType.Exception => "bold red",
+                    _ => "white"
+                };
+                Console.WriteLine($"LOG:[{color}]{logString}[/{color}]");
```
Comment (broarr) [inline unity/CAVE/Assets/Editor/CLI.cs:308]: Do the colors make the input to the python script more difficult to process?

**Review by broarr — APPROVED** [review]
Looks good to me!

File: unity/CAVE/Assets/Editor/CLI.cs (hunk 1 above)
Comment (RobertGemmaJr) [reply unity/CAVE/Assets/Editor/CLI.cs:308]: No, `rich` picks up the syntax automatically so I don't have to worry about it in Python!

Comment (broarr) [reply unity/CAVE/Assets/Editor/CLI.cs:308]: Nice!

File: w3d_translator/w3d_translator.py [hunk 2]
```diff
@@ -10,12 +9,14 @@
...
 UNITY_VERSION = "2021.3.8f1"
 UNITY_PATH = Path(
-    f"C:\\Program Files\\Unity\\Hub\\Editor\\{UNITY_VERSION}\\Editor\\Unity.exe" # noqa
+    f"C:\\Program Files\\Unity\\Hub\\Editor\\{UNITY_VERSION}\\Editor\\Unity.exe"  # noqa
 )
 STARTER_PROJECT = Path("unity/CAVE")
 LOG_FLAG = "LOG:"  # Flag to send prints from the CLI script onto the console
 console = Console()
-err_console = Console(stderr=True)
+err_console = Console(stderr=True, style="bold red")
```
Comment (kmilo9999) [inline w3d_translator/w3d_translator.py:17]: Is this to log the events in the python script?  If that's the case, why not use [logging](https://docs.python.org/3/library/logging.html)

File: w3d_translator/w3d_translator.py [hunk 3]
```diff
@@ -54,111 +55,127 @@ def create_out(out_dir: Path, force: bool):
...
+            ":white_check_mark: "
+            + f"[cyan]{unity_dir.name}/ "
+            + "translated successfully[/cyan]\n"
+        )
     else:
-        console.print(f"'{filename}' is valid")
+        console.print(
+            f"[yellow]{unity_dir.name}/ "
+            + "translated with some errors [/yellow]\n"
+        )
```
Comment (kmilo9999) [inline w3d_translator/w3d_translator.py:103]: Do you really need the `all_good`  variable?

**Review by kmilo9999 — COMMENTED** [review]
Just a few questions on my side.

File: w3d_translator/w3d_translator.py (hunk 2 above)
Comment (RobertGemmaJr) [reply w3d_translator/w3d_translator.py:17]: I've been liking [rich](https://rich.readthedocs.io/en/stable/introduction.html) for it's various, some of which I think will come in really handy once we start multithreading. I'll definitely take a look at logging though! We may be able to use both

File: w3d_translator/w3d_translator.py (hunk 3 above)
Comment (RobertGemmaJr) [reply w3d_translator/w3d_translator.py:103]: When a specific file throws an exception in Unity the file gets skipped and that scene is not saved. But the python script does not crash. I want a way to tell the user that some of the files could not be translated at the end. Perhaps I could keep a running count of how many fail and how many translate correctly?

**Pull request merged**